packages/@aws-cdk/aws-redshift/lib/parameter-group.ts

```diff
@@ -62,17 +63,46 @@ export class ClusterParameterGroup extends ClusterParameterGroupBase {
    */
   public readonly clusterParameterGroupName: string;
 
+  /**
+   * The parameters in the parameter group
+  */
+  readonly parameters: { [name: string]: string };
+
+  /**
+   * The underlying CfnClusterParameterGroup
+  */
+  private readonly resource: CfnClusterParameterGroup;
+
   constructor(scope: Construct, id: string, props: ClusterParameterGroupProps) {
     super(scope, id);
-
-    const resource = new CfnClusterParameterGroup(this, 'Resource', {
+    this.parameters = props.parameters;
+    this.resource = new CfnClusterParameterGroup(this, 'Resource', {
       description: props.description || 'Cluster parameter group for family redshift-1.0',
       parameterGroupFamily: 'redshift-1.0',
-      parameters: Object.entries(props.parameters).map(([name, value]) => {
-        return { parameterName: name, parameterValue: value };
-      }),
+      parameters: this.parseParameters(),
     });
 
-    this.clusterParameterGroupName = resource.ref;
+    this.clusterParameterGroupName = this.resource.ref;
+  }
+  private parseParameters(): any {
+    return Object.entries(this.parameters).map(([name, value]) => {
+      return { parameterName: name, parameterValue: value };
+    });
+  }
```

Comment:
I think instead of allowing `{ [name: string]: string }` input, this should be more strongly typed. If you create a type
```
interface ClusterParameter {
  readonly parameterName: string;
  readonly parameterValue: string;
}
```
and then have these functions take in `parameters: ClusterParameter[]`. Line 42 in this file should also be updated to tighten the contract. 

You can name them differently if you have a better idea for naming. I'm certainly not tied to the specific ones I suggested.

Reply:
Personally I think it makes a less confusing API  to call add parameters multiple times like this (especially in other languages)

```ts
 public addParameter(name: string, value: string): void {}

parameterGroup.addParameter("foo", "bar")
parameterGroup.addParameter("lorem", "ipsum")
parameterGroup.addParameter("key", "key")
```

than 

```ts
  public addParameter(params: ClusterParameter[]): void {}

parameterGroup.addParameter(
    [
        {name: 'foo', value: 'bar'},
        {name: 'lorem', value: 'ipsum'},
        {name: 'key', value: 'key'},
    ]
)
```

or 

```ts
public addParameter(...params: ClusterParameter[]): void {}

parameterGroup.addParameter(
    {name: 'foo', value: 'bar'},
    {name: 'lorem', value: 'ipsum'},
    {name: 'key', value: 'key'}
)
```